flexget/utils/titles/series.py

```diff
@@ -408,31 +405,76 @@ def parse(self, data=None, field=None, quality=None):
                             self.proper_count = int(match.group('version')) - 1
                     self.id_type = 'sequence'
                     self.valid = True
+                    matched_string = match.group(0)
                     log.debug('found id \'%s\' with regexp \'%s\'', self.id, sequence_re.pattern)
                     if not (self.special and self.prefer_specials):
-                        return
+                        found_identifier = True
+                        break
                     else:
                         break
             else:
                 log.debug('-> no luck with sequence_regexps')
 
         # No id found, check if this is a special
-        if self.special or self.assume_special:
+        if not found_identifier and self.special or self.assume_special:
             # Attempt to set id as the title of the special
             self.id = data_stripped or 'special'
             self.id_type = 'special'
             self.valid = True
             log.debug('found special, setting id to \'%s\'', self.id)
-            return
+            found_identifier = True
         if self.valid:
-            return
+            found_identifier = True
 
-        msg = 'Title `%s` looks like series `%s` but cannot find ' % (self.data, self.name)
-        if self.identified_by == 'auto':
-            msg += 'any series numbering.'
-        else:
-            msg += 'a(n) `%s` style identifier.' % self.identified_by
-        raise ParseWarning(self, msg)
+        if not found_identifier:
+            msg = 'Title `%s` looks like series `%s` but cannot find ' % (self.data, self.name)
+            if self.identified_by == 'auto':
+                msg += 'any series numbering.'
+            else:
+                msg += 'a(n) `%s` style identifier.' % self.identified_by
+            raise ParseWarning(self, msg)
+
+        # Remove episode id
+        if found_identifier and matched_string:
+            groups_data_stripped = re.sub(matched_string.replace(' ', '.*?'), '', groups_data_stripped)
+
+        # Find quality and clean from data
```

Comment:
This method has grown beyond hope, consider extracting all the "# do stuff" segments into own methods